### ISSUE TITLE
Periodic "Lua: error in error handling" crashes

### ISSUE THREAD

**Issue by cornernote** [minetest/minetest#3011]
```
2015-08-05 18:44:50: ERROR[main]: UNRECOVERABLE error occurred. Stopping server. Please fix the following error:
2015-08-05 18:44:50: ERROR[main]: Lua: error in error handling

In thread 7f2241deb780:
/home/graeme/skyblock/src/server.cpp:511: void Server::step(float): A fatal error occurred: Lua: error in error handling
Debug stacks:
DEBUG STACK FOR THREAD 7f2239a37700:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f223a238700:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD 7f223aa39700:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
(Leftover data: #2  void Server::SendBlocks(float))
(Leftover data: #3  void Server::SendBlockNoLock(irr::u16, MapBlock*, irr::u8, irr::u16))
(Leftover data: #4  void ItemStack::serialize(std::ostream&) const)
(Leftover data: #5  bool getCraftingResult(Inventory*, ItemStack&, std::vector<ItemStack>&, bool, IGameDef*))
(Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f2241deb780:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
#3  void Server::step(float)
(Leftover data: #4  void Server::SendAccessDenied_Legacy(irr::u16, const wstring&))
```

one here:
https://github.com/cornernote/minetest-skyblock/issues/28

and another one here:
http://hastebin.com/raw/dexaxowusu

**Comment by kwolekr** [comment]
Try updating to the latest version.

**Comment by thatgraemeguy** [comment]
Updating now. Any idea how I would know if its fixed? This happened at least half a dozen times and every time there seemed to be no common preceding log entry indicating an action that caused it.

**Comment by cornernote** [comment]
seemed to be when running the code registered by `minetest.register_on_craft`

if it happens frequently then just run the server and see if it happens.  If all good for a couple of days then please report your findings back here

**Comment by thatgraemeguy** [comment]
Just happened again, or is this a new, unrelated thing?

    2015-08-06 15:30:57: ACTION[ServerThread]: thatgraemeguy places node default:torch at (1607,3,1604)
    2015-08-06 15:30:59: ACTION[ServerThread]: Blasus places node default:desert_sand at (1727,5,1566)
    2015-08-06 15:31:00: ERROR[main]: UNRECOVERABLE error occurred. Stopping server. Please fix the following error:
    2015-08-06 15:31:00: ERROR[main]: Lua: Double fault error in luaentity_Step(): error in error handling

    In thread 7f19d8cab780:
    /home/graeme/skyblock/src/server.cpp:511: void Server::step(float): A fatal error occurred: Lua: Double fault error in luaentity_Step(): error in error handling
    Debug stacks:
    DEBUG STACK FOR THREAD 7f19d08f7700:
    #0  virtual void* EmergeThread::Thread()
    (Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
    (Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
    (Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
    DEBUG STACK FOR THREAD 7f19d10f8700:
    #0  virtual void* CurlFetchThread::Thread()
    DEBUG STACK FOR THREAD 7f19d18f9700:
    #0  virtual void* ServerThread::Thread()
    #1  void Server::Receive()
    (Leftover data: #2  void Server::SendBlocks(float))
    (Leftover data: #3  void RemoteClient::GetNextBlocks(ServerEnvironment*, EmergeManager*, float, std::vector<PrioritySortedBlockTransfer>&))
    (Leftover data: #4  void ItemStack::serialize(std::ostream&) const)
    (Leftover data: #5  bool getCraftingResult(Inventory*, ItemStack&, std::vector<ItemStack>&, bool, IGameDef*))
    (Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
    (Leftover data: #7  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
    DEBUG STACK FOR THREAD 7f19d8cab780:
    #0  int main(int, char**)
    #1  Dedicated server branch
    #2  void dedicated_server_loop(Server&, bool&)
    #3  void Server::step(float)
    (Leftover data: #4  void Server::SendAccessDenied_Legacy(irr::u16, const wstring&))

**Comment by kwolekr** [comment]
@thatgraemeguy Yes, that is the same error.  Yours happened inside of a minetest.register_entity on_step() callback.

**Comment by thatgraemeguy** [comment]
Taken from my world.mt:

    load_mod_3d_armor = true
    load_mod_moreblocks = true
    load_mod_protector = true
    load_mod_shields = true
    load_mod_skyblock_feats = true
    load_mod_skyblock_levels = true
    load_mod_skyblock = true
    load_mod_stairsplus = true
    load_mod_travelnet = true
    load_mod_unified_inventory = true
    load_mod_wieldview = true

If there's any other info that will help let me know. I have all server's console output logged. Implemented a cron to check if it ought to be running and isn't, which has triggered 20+ times over the last 11 or so hours.

**Comment by cornernote** [comment]
on OldCoders server:

```
beds
blox
boats
bones
bookex
bucket
creative
decoplants
default
doors
dye
farming
fire
flowers
gloopblocks
ilights
irc
irc_commands
moreflags
news
player_textures
protector
screwdriver
sethome
skyblock
stairs
throwing
tnt
travelnet
unified_inventory
vessels
wiki
wool
worldedit-modpack
xban2
xpanes
```

**Comment by HybridDog** [comment]
l found a way to reproduce the error in error handling:
https://gist.github.com/HybridDog/1f868a012b1f59cdf00c#file-crash_mod_file-lua

**Comment by thatgraemeguy** [comment]
My server has been running for nearly 24 hours without issue.

**Comment by tchncs** [comment]
Same Problem here with the following Mods:

```
3d_armor     
darkage            
mesecons    
moreores   
plantlife_modpack  
stats      
unifieddyes
arrow_signs  
farming_plus       
mini_sun    
moretrees  
quartz             
steel
boost_cart   
homedecor_modpack  
mobs        
mr_goat    
random_buildings   
the_list
cannons      
kickall            
mobs_sky    
mushrooms  
scaffolding        
torches
craft_guide  
medieval_craft     
moreblocks  
pipeworks  
stairsplus         
travelnet
```
I think the Minetest Client should have a Feature to check if the Server is up.

**Comment by est31** [comment]
Note: the scope of this bug is now to track "ocasionally my server throws this 'error in error handling message'", if #3024 is resolved, and this bug still appears, we can find out whether its engine fault, or modder fault. If its modder fault, this bug can be closed.

**Comment by kwolekr** [comment]
Well, I guess it must be travelnet.

**Comment by est31** [comment]
Seen here too: https://forum.minetest.net/viewtopic.php?f=6&t=12962

**Comment by kwolekr** [comment]
@thatgraemeguy I am having trouble reproducing this error but you seem to have them quite regularly.  What is the best way to get in touch with you?

**Comment by thatgraemeguy** [comment]
I'm on IRC most of the time, Freenode and Inchra. Unfortunately I don't seem to be getting it any more though.

    <ThatGraemeGuy> [08:36:05] hmmmm: FWIW, I haven't had "error in error handling" happen again for 3 days now
    <ThatGraemeGuy> [08:36:41] and while I had travelnet enabled, while that error was occurring, nobody had built any travelnets yet
    <ThatGraemeGuy> [08:37:23] its a newish skyblock world, so far there are only 2 travelnet boxes and they were built after the error stopped occurring

**Comment by est31** [comment]
I think this is a blocker issue because multiple people reported it, and had it very often.

**Comment by HybridDog** [comment]
maybe error handling doesn't work in specific situations, e.g. digging and placing

**Comment by est31** [comment]
@HybridDog thats what #3024 is about, this issue tracks error in error handling errors that are the engine's fault.

**Comment by kwolekr** [comment]
Removed blocker status because frankly it's too hard to track down and for some reason not many people seem to be having it any more.  The frequency of the errors appears to periodically wax and wane.  Probably due to some corrupted mod state being reset or the offending object removed.

**Comment by est31** [comment]
@kwolekr has said it was likely the travelnet mod by sokomine, and not the engine, therefore closing this, as people also don't report it anymore. It seems to be not the engine's fault, that's what matters.

**Comment by tchncs** [comment]
We have this problem with mobf_trader mod by sokomine also... :-(

**Comment by amadin** [comment]
I also sometimes have "error in error handling" on 0.4.13 stable.

**Comment by est31** [comment]
The fix isn't in yet in 0.4.13 stable.
